### ISSUE TITLE
never return a result to the client broswer when I kill the request thread in code before the execution of method is finished 

### ISSUE THREAD

**Issue by w281015308** [NancyFx/Nancy#2950]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of Nancy
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/NancyFx/Nancy/issues) and [closed](https://github.com/NancyFx/Nancy/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

 
![image](https://user-images.githubusercontent.com/11220995/51022426-25492c00-15bf-11e9-9667-0ab7444878d3.png)

![image](https://user-images.githubusercontent.com/11220995/51022859-5a09b300-15c0-11e9-9d39-198ca539ffb9.png)

![image](https://user-images.githubusercontent.com/11220995/51023046-da301880-15c0-11e9-87ec-f3a98fa4b28d.png)

but the codes I added just for testing. It is not really correct.
 

### System Configuration

- Nancy version: latest version
  
- Environment (Operating system, version and so on): windows 10
- .NET Framework version:  .net 4.5.2

**Comment by robertbissonnette** [comment]
I gotta ask. What exactly is your use case with this?

Normal operation would not have you killing threads in the context of the server to begin with.  But in your scenario you are actually killing threads that below to the thread pool.  If you want to terminate the execution and return a response to the caller, then you should either use an async model with a cancellationToken or throw an exception.

**Comment by w281015308** [comment]
> I gotta ask. What exactly is your use case with this?
> 
> Normal operation would not have you killing threads in the context of the server to begin with. But in your scenario you are actually killing threads that below to the thread pool. If you want to terminate the execution and return a response to the caller, then you should either use an async model with a cancellationToken or throw an exception.

Yes, I really want to terminate the execution. I have a circuit breaker to limit the concurrent request  and it is working in another thread. when the threshold value of concurrent request is reached, circuit breaker 
 can not throw an exception in the request thread. so kill the request thread is simple. Cause our scenario is very complex, I have to do that. 
If I change Nancy code for throwing an exception when the request thread is killed, It is correct or not?
Thanks for your reply. :)
